src/Workspaces/Core/Portable/Execution/Serialization/SnapshotBuilder.cs

```diff
@@ -41,22 +41,22 @@ private Task<ChecksumCollection> BuildAsync(Solution solution, IEnumerable<Proje
 
         private Task<ChecksumCollection> BuildAsync(Project project, IEnumerable<TextDocument> documents, string kind, CancellationToken cancellationToken)
         {
-            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.GetProjectState(project.Id), documents, kind, CreateDocumentChecksumObjectAsync, cancellationToken);
+            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.State.GetProjectState(project.Id), documents, kind, CreateDocumentChecksumObjectAsync, cancellationToken);
         }
 
         private Task<ChecksumCollection> BuildAsync(Project project, IEnumerable<ProjectReference> projectReferences, CancellationToken cancellationToken)
         {
-            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.GetProjectState(project.Id), projectReferences, WellKnownChecksumObjects.ProjectReferences, CreateChecksumCollectionsAsync, cancellationToken);
+            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.State.GetProjectState(project.Id), projectReferences, WellKnownChecksumObjects.ProjectReferences, CreateChecksumCollectionsAsync, cancellationToken);
         }
 
         private Task<ChecksumCollection> BuildAsync(Project project, IEnumerable<MetadataReference> metadataReferences, CancellationToken cancellationToken)
         {
-            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.GetProjectState(project.Id), metadataReferences, WellKnownChecksumObjects.MetadataReferences, CreateChecksumCollectionsAsync, cancellationToken);
+            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.State.GetProjectState(project.Id), metadataReferences, WellKnownChecksumObjects.MetadataReferences, CreateChecksumCollectionsAsync, cancellationToken);
         }
 
         private Task<ChecksumCollection> BuildAsync(Project project, IEnumerable<AnalyzerReference> analyzerReferences, CancellationToken cancellationToken)
         {
-            return _checksumTree.GetOrCreateChecksumObjectWithChildrenAsync(project.Solution.GetProjectState(project.Id), analyzerReferences, WellKnownChecksumObjects.AnalyzerReferences, CreateChecksumCollectionsAsync, cancellationToken);
```

Comment:
I can. couldn't decide what would be better. since when I was refactoring, it was confusing what is public API of solution or internal API only us can use.